Ubuntu24.04環境でのビルドエラー対応

<!--
***********************************************************************
*                           << NOTE >>                                *
*  This repository have been programmed in C++.                       *
*                                                                     *
*  Please do not create tickets for other programming languages here. *
*  If you will create tickets for other programming language,         *
*  please see below:                                                  *
*                                                                     *
*  - Python : https://github.com/OpenRTM/OpenRTM-aist-Python/issues   *
*  - Java   : https://github.com/OpenRTM/OpenRTM-aist-Java/issues     *
*                                                                     *
***********************************************************************
-->

**Describe the bug**
Ubuntu24.04環境でC++のビルドが通らない。2種類のエラーが発生する。

- 最初にcold属性のエラーが出る
  ```
  OpenRTM-aist/src/lib/coil/common/coil/Logger.cpp:286:5: error: function might be candidate for attribute ‘cold’ [-Werror=suggest-attribute=cold]
  ```
  - Ubuntu24.04のgccバージョンが13.2.0のためで、一番トップのCMakeLists.txtで定義している「-Wsuggest-attribute=cold」でのバージョンを13.9.9にすれば解消できる

- 上記エラー対応をすると、次に下記エラーが発生する
  ```
  OpenRTM-aist/src/lib/rtm/Manager.cpp:1231:60: error: redundant move in initialization [-Werror=redundant-move]
   1231 |         std::string mpm_{coil::eraseBothEndsBlank(std::move(itr))};
        |                                                   ~~~~~~~~~^~~~~
  ```
  - これは以下のように修正すると解消する
    ```
    std::string mpm_{coil::eraseBothEndsBlank(std::move(itr))};
        ↓
    std::string mpm_{coil::eraseBothEndsBlank(itr)};
    ```
    - 参考情報： https://lists.yoctoproject.org/g/meta-arm/message/4395

**Reproducibility**
100%


**Environment**
 - OS: Ubuntu 24.04
 - OpenRTM-aist Version 2.0.2

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -688,7 +688,7 @@ function(openrtm_common_set_compile_options target)
 		-Wstrict-null-sentinel
 		-Wstrict-overflow=5
 		$<$<VERSION_GREATER:${CMAKE_CXX_COMPILER_VERSION},6.9.9>: -Wstringop-overflow=4>
-		$<$<VERSION_GREATER:${CMAKE_CXX_COMPILER_VERSION},12.9.9>: -Wsuggest-attribute=cold>
+		$<$<VERSION_GREATER:${CMAKE_CXX_COMPILER_VERSION},13.9.9>: -Wsuggest-attribute=cold>
 		-Wsuggest-attribute=format
 		$<$<VERSION_GREATER:${CMAKE_CXX_COMPILER_VERSION},7.9.9>: -Wsuggest-attribute=malloc>
 		$<$<VERSION_GREATER:${CMAKE_CXX_COMPILER_VERSION},4.9.9>: -Wsuggest-override>

--- a/src/lib/rtm/Manager.cpp
+++ b/src/lib/rtm/Manager.cpp
@@ -1228,7 +1228,7 @@ std::vector<coil::Properties> Manager::getLoadableModules()
 
     for (auto const& itr : coil::split(m_config["manager.preload.modules"], ","))
       {
-        std::string mpm_{coil::eraseBothEndsBlank(std::move(itr))};
+        std::string mpm_{coil::eraseBothEndsBlank(itr)};
         if (mpm_.empty())
           {
               continue;

